snap7/client.py

```diff
@@ -389,9 +389,11 @@ def read_area(self, area: Areas, dbnumber: int, start: int, size: int) -> bytear
             >>> buffer
             bytearray(b'\\x00\\x00')
         """
-        if area not in Areas:
-            raise ValueError(f"{area} is not implemented in snap7.types")
-        elif area == Areas.TM:
+        try:
+            area = Areas(area)
```

Comment:
This works if `area` is an `Area` object already? 

If we do want to add support for both `int` and `Area` types I think we should update the function signature to accept `area: Union[Area, int]` (or `area: Area | int` for py3.10 but we support older versions).

Reply:
>This works if area is an Area object already?

Yes, it works

>If we do want to add support for both int and Area 

We don't need to support `int` type and we shouldn't. Let's call it undocumented feature. 

Maybe i'm wrong, but in my opinion, `Areas` is a bunch of integer constants, with `area = Areas(area)` we just make sure the `area` value is one of them. If it works with integer - good for lazy people like me.

Reply:
I'm surprised mypy is not complaining since it is wrongly typed then? We run mypy on the test suite also, and it should raise an error then since int is not a valid argument for the function according to the typing.

I'm not sure if it is generally a good idea to ignore the type system; it helps find bugs while doing static code analysis. 

If the type checking would work correctly, a user is using mypy and this user wants to insert an int into this function he has to disable the type-checking of this function. So if we do add support for both types maybe it is best to be clear about that, and in the function check the type and act accordingly. The static code compiler probably understands that also.

Reply:
Pycharm complains, i didn't figure out yet why mypy doesn't complain:

![Schermafbeelding 2022-01-20 om 09 26 34](https://user-images.githubusercontent.com/326308/150292556-78de70cb-e7be-48ed-beb0-e1b5147c00a4.png)

Reply:
If mypy against me I have no choice but surrender. I was hoping that mypy regards IntEnum as a subclass of int, but it doesn't


![image](https://user-images.githubusercontent.com/52915342/151850473-e355a6b6-5208-4b17-9290-13125805421c.png)